src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
                     ErrorsFound = true;
-                } else if (PurchAir(PurchAirNum).DCVType == DCV::CO2SetPoint && !state.dataContaminantBalance->Contaminant.CO2Simulation) {
-                    PurchAir(PurchAirNum).DCVType = DCV::None;
-                    ShowWarningError(state, format("{}{}=\"{} invalid data", RoutineName, s_ipsc->cCurrentModuleObject, s_ipsc->cAlphaArgs(1)));
-                    ShowContinueError(state, format("{}={} but CO2 simulation is not active.", s_ipsc->cAlphaFieldNames(14), s_ipsc->cAlphaArgs(14)));
-                    ShowContinueError(state, format("Resetting {} to NoDCV", s_ipsc->cAlphaFieldNames(14)));
-                    ShowContinueError(state,
-                                      "To activate CO2 simulation, use ZoneAirContaminantBalance object and specify \"Carbon Dioxide "
-                                      "Concentration\"=\"Yes\".");
                 }
 
                 // get Outdoor air economizer type
-                PurchAir(PurchAirNum).EconomizerType = static_cast<Econ>(getEnumValue(econNamesUC, s_ipsc->cAlphaArgs(15)));
-                if (PurchAir(PurchAirNum).EconomizerType == Econ::Invalid) {
-                    ShowSevereInvalidKey(state,
-                                         eoh,
-                                         s_ipsc->cAlphaFieldNames(15),
-                                         s_ipsc->cAlphaArgs(15),
-                                         "Valid entries are NoEconomizer, DifferentialDryBulb, or DifferentialEnthalpy");
+                cAlphaFieldName = "Outdoor Air Economizer Type";
+                fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "outdoor_air_economizer_type");
+                PurchAir.EconomizerType = static_cast<Econ>(getEnumValue(econNamesUC, Util::makeUPPER(fieldValue)));
+                if (PurchAir.EconomizerType == Econ::Invalid) {
+                    ShowSevereInvalidKey(
+                        state, eoh, cAlphaFieldName, fieldValue, "Valid entries are NoEconomizer, DifferentialDryBulb, or DifferentialEnthalpy.");
                     ErrorsFound = true;
                 }
 
                 // get Outdoor air heat recovery type and effectiveness
-                PurchAir(PurchAirNum).HtRecType = static_cast<HeatRecovery>(getEnumValue(heatRecoveryNamesUC, s_ipsc->cAlphaArgs(16)));
-                if (PurchAir(PurchAirNum).HtRecType == HeatRecovery::Invalid) {
-                    ShowSevereInvalidKey(
-                        state, eoh, s_ipsc->cAlphaFieldNames(16), s_ipsc->cAlphaArgs(16), "Valid entries are None, Sensible, or Enthalpy");
+                cAlphaFieldName = "Heat Recovery Type";
+                fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "heat_recovery_type");
+                PurchAir.HtRecType = static_cast<HeatRecovery>(getEnumValue(heatRecoveryNamesUC, Util::makeUPPER(fieldValue)));
+                if (PurchAir.HtRecType == HeatRecovery::Invalid) {
```

Comment:
Not possible.


    A16, \field Heat Recovery Type
       \type choice
       \key None
       \key Sensible
       \key Enthalpy
       \default None